client/packages/sprotty-client/src/features/layout/layout-commands.ts

```diff
@@ -0,0 +1,416 @@
+    SetBoundsAction,
+    SModelElement,
+    TYPES
+} from "sprotty";
+import { matchesKeystroke } from "sprotty/lib/utils/keyboard";
+
+import { GLSP_TYPES } from "../../types";
+import { isNonRoutableSelectedBoundsAware, SelectableBoundsAware } from "../../utils/smodel-util";
+import { ChangeBoundsOperationAction } from "../operation/operation-actions";
+import { SelectionService } from "../select/selection-service";
+
+export enum ResizeDimension {
+    Width,
+    Height,
+    Width_And_Height
+}
+
+export namespace Reduce {
+    export function min(...values: number[]): number {
+        return Math.min(...values);
+    }
+
+    export function max(...values: number[]): number {
+        return Math.max(...values);
+    }
+
+    export function avg(...values: number[]): number {
+        return values.reduce((a, b) => a + b, 0) / values.length;
+    }
+
+    export function first(...values: number[]): number {
+        return values[0];
+    }
+
+    export function last(...values: number[]): number {
+        return values[values.length - 1];
+    }
+}
+
+export class ResizeElementsAction implements Action {
+    readonly kind = ResizeElementsCommand.KIND;
+
+    constructor(
+        /**
+         * IDs of the elements that should be resized. If no IDs are given, the selected elements will be resized.
+         */
+        public readonly elementIds: string[] = [],
+        /**
+         * Resize dimension.
+         */
+        public readonly dimension: ResizeDimension = ResizeDimension.Width,
+        /**
+         * Function to reduce the dimension to a target dimension value, see Reduce.* for potential functions.
+         */
+        public readonly reductionFunction: (...values: number[]) => number) {
+    }
+}
+
+export enum Alignment {
+    Left,
+    Center,
+    Right,
+    Top,
+    Middle,
+    Bottom
+}
+
+export namespace Select {
+    export function all(elements: SelectableBoundsAware[]) {
+        return elements;
+    }
+
+    export function first(elements: SelectableBoundsAware[]) {
+        return [elements[0]];
+    }
+
+    export function last(elements: SelectableBoundsAware[]) {
+        return [elements[elements.length - 1]];
+    }
+}
+
+
+export class AlignElementsAction implements Action {
+    readonly kind = AlignElementsCommand.KIND;
+
+    constructor(
+        /**
+         * IDs of the elements that should be aligned. If no IDs are given, the selected elements will be aligned.
+         */
+        public readonly elementIds: string[] = [],
+        /**
+         * Alignment direction.
+         */
+        public readonly alignment: Alignment = Alignment.Left,
+        /**
+         * Function to selected elements that are considered during alignment calculation, see Select.* for potential functions.
+         */
+        public readonly selectionFunction: (elements: SelectableBoundsAware[]) => SelectableBoundsAware[] = Select.all) {
+    }
+}
+
+abstract class LayoutElementsCommand extends Command {
```

Comment:
According to sprotty guidelines, commands should be annotated with `@injectable`

Reply:
Done